### PR TITLE
Remove old js components: Card, Table

### PR DESCRIPTION
**Pull request by antgamdia**
## Description of the change

This PR is following up #6554 (and requires it to get merged first). In this final one, the focus is on:

- Transform the `Card` and its sub-components to tsx.
- Transform the `Table` to tsx
- Remove the remaining test snapshots 

### Benefits

No cool before/after pics this time :P  Here we have just migrated js-based code to typescript.
However, after this PR we will able to say we finally got rid of the legacy js-components!! 

### Possible drawbacks

N/A - hopefully.

### Applicable issues

Related to https://github.com/vmware-tanzu/kubeapps/issues/6563

### Additional information

While transforming the `Card` to tsx I also had a look at the `CdsCard` from Clarity. Unfortunately, there is no drop-in replacement; a migration would require plenty of style-related changes. Besides, the "native" style differs a bit from the general style we have in Kubeapps. Below are some examples:

<details><summary>Example from the Clarity website</summary>

image

</details>

<details><summary>Quick/dirty replacement, it does not look good</summary>

image

</details>

<details><summary>Our current custom card</summary>

image

</details>

### CODE REVIEW

File: dashboard/src/components/InfoCard/Card/CardImage/CardImage.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+const CardImage = ({ src, alt }: ICardImageProps) => {
+  return (
+    <div className="card-img">
+      <img src={src} alt={alt} />
+    </div>
+  );
+};
+
+export default CardImage;
```
Comment (absoludity) [inline dashboard/src/components/InfoCard/Card/CardImage/CardImage.tsx:17]: Almost doesn't seem worth a separate component?

File: dashboard/src/components/InfoCard/Card/CardText/CardText.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export interface ICardTextProps {
+  children: React.ReactNode;
+}
+
+const CardText = ({ children }: ICardTextProps) => {
+  return <div className="card-text">{children}</div>;
+};
+
+export default CardText;
```
Comment (absoludity) [inline dashboard/src/components/InfoCard/Card/CardText/CardText.tsx:14]: Same here. Is it a react thing to break the elements up like this? Seems like a `/InfoCard/Card` could just have the text and image without splitting those out into separate components?

**Review by absoludity — APPROVED** [review]
Sorry, I'd missed that this one was still waiting for a review.

File: dashboard/src/components/InfoCard/Card/CardText/CardText.tsx (hunk 2 above)
Comment (antgamdia) [reply dashboard/src/components/InfoCard/Card/CardText/CardText.tsx:14]: 100% agree, no actual benefits, especially if we ever plan to use the card from clarity at some point in the future. I thought the same while moving the code to tsx. I'll make a follow-up PR with the changes shortly.

**Pull request merged**